File: groups/group7/java-version-original/DivisionAndEuclidean/DivisionAlgorithm.java
import java.util.Scanner;
import java.util.InputMismatchException;

public class DivisionAlgorithm {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.println("\n===== DIVISION ALGORITHM =====");

            int a = readPositiveInt(scanner, "Enter the first integer: ");
            int b = readPositiveInt(scanner, "Enter the second integer: ");

            // Higher integer = dividend (m), lower = divisor (n)
            int m = Math.max(a, b);
            int n = Math.min(a, b);

            int q = m / n;
            int r = m % n;

            System.out.println("\nSOLUTION:");
            // Format: "m = n(q) + r"  — no space before '(' per activity spec
            System.out.println(m + " = " + n + "(" + q + ") + " + r);
            System.out.println("The dividend is " + m);
            System.out.println("The divisor is " + n);
            System.out.println("The quotient is " + q + " and the remainder is " + r);

            System.out.print("\nCompute again? (y/n): ");
            String choice = scanner.nextLine().trim().toLowerCase();
            if (!choice.equals("y")) {
                System.out.println("Goodbye!");
                break;
            }
        }

        scanner.close();
    }

    /**
     * Reads a strictly positive integer from the user.
     * Handles: letters, symbols, decimals, negatives, zero, overflow.
     */
    static int readPositiveInt(Scanner scanner, String prompt) {
        while (true) {
            System.out.print(prompt);
            String input = scanner.nextLine().trim();

            if (input.isEmpty()) {
                System.out.println("  ERROR: Input cannot be empty. Please enter a positive integer.");
                continue;
            }

            // Reject decimals explicitly before parseInt swallows them as exceptions
            if (input.contains(".")) {
                System.out.println("  ERROR: Decimals are not allowed. Please enter a whole number.");
                continue;
            }

            // Reject anything that isn't an optional '-' or '+' followed by digits
            if (!input.matches("[+-]?\\d+")) {
                System.out.println("  ERROR: \"" + input + "\" is not a valid integer. No letters or symbols allowed.");
                continue;
            }

            int value;
            try {
                value = Integer.parseInt(input);
            } catch (NumberFormatException e) {
                System.out.println("  ERROR: Number is too large. Please enter a value within the int range.");
                continue;
            }

            if (value <= 0) {
                System.out.println("  ERROR: Only positive integers (greater than 0) are accepted.");
                continue;
            }

            return value;
        }
    }
}

File: groups/group9/DivisionAlgorithm.js
function main(inputs) {
  const raw1 = String(inputs[0] ?? "").trim();
  const raw2 = String(inputs[1] ?? "").trim();
  if (!raw1 || !raw2)
    throw new Error("INVALID OUTPUT\nPlease complete all required inputs.");
  const first = parseInt(raw1, 10);
  const second = parseInt(raw2, 10);
  if (isNaN(first) || !Number.isInteger(Number(raw1)))
    throw new Error("INVALID OUTPUT\nFirst input must be a whole number.");
  if (isNaN(second) || !Number.isInteger(Number(raw2)))
    throw new Error("INVALID OUTPUT\nSecond input must be a whole number.");
  if (first <= 0)
    throw new Error("INVALID OUTPUT\nFirst input must be a positive integer.");
  if (second <= 0)
    throw new Error("INVALID OUTPUT\nSecond input must be a positive integer.");

  const m = Math.max(first, second);
  const n = Math.min(first, second);
  const q = Math.floor(m / n);
  const r = m % n;

  return [
    "SOLUTION:",
    `${m} = ${n} (${q}) + ${r}`,
    ``,
    `The dividend is ${m}`,
    `The divisor is ${n}`,
    `The quotient is ${q} and the remainder is ${r}`
  ].join("\n");
}